ios/browser/api/sync/prefs/brave_sync_profile_service.mm

```diff
@@ -0,0 +1,125 @@
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#import "brave/ios/browser/api/sync/prefs/brave_sync_profile_service.h"
+
+#include <unordered_map>
+
+#include "components/sync/base/model_type.h"
+#include "components/sync/base/user_selectable_type.h"
+#include "components/sync/driver/sync_service.h"
+#include "components/sync/driver/sync_user_settings.h"
+#include "ios/chrome/browser/application_context.h"
+#include "ios/chrome/browser/browser_state/chrome_browser_state.h"
+#include "ios/chrome/browser/sync/profile_sync_service_factory.h"
+#include "ios/web/public/thread/web_thread.h"
+
+#if !defined(__has_feature) || !__has_feature(objc_arc)
+#error "This file requires ARC support."
+#endif
+
+namespace brave {
+namespace ios {
+std::unordered_map<syncer::UserSelectableType, BraveSyncUserSelectableTypes>
+    mapping = {
+        {syncer::UserSelectableType::kBookmarks,
+         BraveSyncUserSelectableTypes_BOOKMARKS},
+        {syncer::UserSelectableType::kPreferences,
+         BraveSyncUserSelectableTypes_PREFERENCES},
+        {syncer::UserSelectableType::kPasswords,
+         BraveSyncUserSelectableTypes_PASSWORDS},
+        {syncer::UserSelectableType::kAutofill,
+         BraveSyncUserSelectableTypes_AUTOFILL},
+        {syncer::UserSelectableType::kThemes,
+         BraveSyncUserSelectableTypes_THEMES},
+        {syncer::UserSelectableType::kHistory,
+         BraveSyncUserSelectableTypes_HISTORY},
+        {syncer::UserSelectableType::kExtensions,
+         BraveSyncUserSelectableTypes_EXTENSIONS},
+        {syncer::UserSelectableType::kApps, BraveSyncUserSelectableTypes_APPS},
+        {syncer::UserSelectableType::kReadingList,
+         BraveSyncUserSelectableTypes_READING_LIST},
+        {syncer::UserSelectableType::kTabs, BraveSyncUserSelectableTypes_TABS},
+        {syncer::UserSelectableType::kWifiConfigurations,
+         BraveSyncUserSelectableTypes_WIFI_CONFIGURATIONS}};
+
+syncer::UserSelectableTypeSet user_types_from_options(
+    BraveSyncUserSelectableTypes options) {
+  syncer::UserSelectableTypeSet results;
+  for (auto it = mapping.begin(); it != mapping.end(); ++it) {
+    if (options & it->second) {
+      results.Put(it->first);
+    }
+  }
+  return results;
+}
+
+BraveSyncUserSelectableTypes options_from_user_types(
+    const syncer::UserSelectableTypeSet& types) {
+  BraveSyncUserSelectableTypes results = BraveSyncUserSelectableTypes_NONE;
+  for (auto it = mapping.begin(); it != mapping.end(); ++it) {
+    if (types.Has(it->first)) {
+      results |= it->second;
+    }
+  }
+  return results;
+}
+}  // namespace ios
+}  // namespace brave
+
+@interface BraveSyncProfileService () {
+  syncer::SyncService* sync_service_;
+  std::unordered_map<syncer::UserSelectableType, BraveSyncUserSelectableTypes>
+      type_mapping;
+}
+@end
+
+@implementation BraveSyncProfileService
+
+- (instancetype)initWithBrowserState:(ChromeBrowserState*)state {
```

Comment:
it's actually better to have `initWithProfileSyncService` here since we don't actually need `ChromeBrowserState`